### ISSUE TITLE
[LINT_BUG]: Error when searching for module which uses ../

### ISSUE THREAD

**Issue by tyner** [Appsilon/box.linters#137]
### box.linters version

0.10.2

### Sample source code to lint

Somewhat similar to #110 but this time using ../

Create a file `Foo.R` containing:
``` r
foo <- function(x) {
    box::use(../Bar[baz])

    baz(x)
}
```

Then in the parent directory of the directory where `Foo.R` lives, create a file `Bar.R` containing:
``` r
baz <- function(x) {
    message(x, " from ", sys.call()[[1L]])
}
```

(NB: I omitted # 'export directives above because those were not playing nice with Markdown).

then run the following to verify that the example is a valid setup from the perspective of `box` mechanics
``` r
options(box.path = "/path/to/directory") # adjust this to your actual test directory
box::use(./Foo[foo])
foo("hello")
```
(it prints out `hello from baz` as a confirmation)

### Lint command used

```r 
lintr::lint(filename = "/path/to/directory/Foo.R", # adjust this to your actual test directory
            linters = box.linters::box_default_linters
            )
```

I tried various strategies to get it to work, including setting `options(box.path)` as well as the working directory.

### Lint result

Error: Linter 'box_mod_fun_exists_linter' failed in /path/to/directory/Foo.R: unable to load module “../Bar”; not found

### Expected result

Would be nice for it to find the module. Perhaps this is not supported, but I didn't see it mentioned in the documentation.

**Comment by TymekDev** [comment]
@tyner I have run into—what I thought—was a similar issue and decided do reproduce your issue. However, I couldn't quite do that.

## Setup 
I created the following structure:
```
.
├── directory
│   └── Foo.R
├── Bar.R
├── lint.R
└── main.R
```
I added `#' @export` before functions in `Foo.R` and `Bar.R`. Otherwise, they are the same as in your comment.

**`lint.R`**
```R
lintr::lint("directory/Foo.R", linters = box.linters::box_default_linters)
```

**`main.R`**
```R
options(box.path = "directory")
box::use(./Foo[foo])
foo("hello")
```

_I am using R 4.2.2 and {box.linters} installed from source 7b25817 (the v0.10.2 tag is missing)._

## Outcome

Running `lint.R` doesn't produce any lints. Running `main.R` errors on import.

https://github.com/user-attachments/assets/f2ace4f9-5b1f-4c75-9da4-d7c4b0896cf5

I played with this a bit more to get a better understanding and I _think_ this is the expected behavior. 

[A Search Path section in `box::use`'s documentation](https://klmr.me/box/reference/use.html#search-path) confirms this (emphasis is mine):

> Modules are searched in **the module search path**, given by getOption('box.path').
> [...]
> Local import declarations (that is, module prefixes that start with `./` or `../`) **never use the search path** to find the module. Instead, only the current module’s directory (for `./`) or the parent module’s directory (for `../`) is looked at.

## When to use `box.path`

In the gist, `box.path` should be used if you wanted to have a `Baz.R` script in `directory/` that would use an absolute import. For example:

**`directory/Baz.R`**
```R
box::use(
  directory/Foo[foo]
)
```

Without setting the `box.path` to the root directory (the `directory/`'s parent), you would get a following error:
```
Error in box::use(directory/Baz) :
  unable to load module “directory/Foo”; not found in “~/Library/R/arm64/4.2/library/box/mod”, “~/work/playground/box.linters/issue-137/directory”
Calls: <Anonymous> ... tryCatchList -> tryCatchOne -> <Anonymous> -> rethrow
Execution halted
```

**Comment by TymekDev** [comment]
There is an issue about mixing `box.path` with relative imports, though.

## Setup

```
.
├── directory
│   ├── foo.R
│   └── main.R
├── lint.R
└── lint_box_path.R
```

**`directory/foo.R`**
```R
#' @export
hello <- function() {
  print("hello world!")
}
```

**`directory/main.R`**
```R
box::use(
  ./foo[hello]
)

hello()
```

**`lint.R`**
```R
lintr::lint("directory/main.R", linters = box.linters::box_default_linters)
```

**`lint_box_path.R`**
```R
options(box.path = getwd())
lintr::lint("directory/main.R", linters = box.linters::box_default_linters)
```

## Outcome

1. `Rscript directory/main.R` - works
2. `Rscript lint.R` - works
3. `Rscript lint_box_path.R` - fails with:
    ```
    Error: Linter 'box_mod_fun_exists_linter' failed in <snip>/directory/main.R: unable to load module “./foo”; not found in “.”
    Execution halted
    ```

## Reason

The helper function for determining the "working directory of a module" uses the `box.path` value if it's set:

https://github.com/Appsilon/box.linters/blob/27df08a0bed9e5576130a344b29c963d8760be2e/R/box_module_usage_helper_functions.R#L186-L194

This is contradicting {box} docs I linked above.

> Modules are searched in the module search path, given by getOption('box.path').
> [...]
> Local import declarations (that is, module prefixes that start with ./ or ../) never use the search path to find the module. Instead, only the current module’s directory (for ./) or the parent module’s directory (for ../) is looked at.

---

I am not sure if this is what you, @tyner, have been getting at or if it's an entirely separate issue.

**Comment by tyner** [comment]
Thank you @TymekDev, your analysis is spot on and I greatly appreciate it. 

Looking back, my initial post was flawed/ambiguous -- I didn't make it clear what the 'test directory' was, and was unaware that `box::use(./Foo[foo])` was entirely bypassing the `box.path` option. 

Also thank you for pointing out that `box.linters:::get_module_working_dir` seems inconsistent in this regard. For now, leaving the thread open, in case further discussion is warranted on that.

**Comment by radbasa** [comment]
A relevant comment made in a different issue https://github.com/Appsilon/box.linters/issues/108#issuecomment-2365165817

**Comment by tyner** [comment]
Tried that same example under version 0.10.5 of box.linters and now it gives an error:

```
Error in !trace_length(trace) : invalid argument type
```
buried in the `traceback()` is

```
message = "unable to load module “../Bar”
```
